l10n_ch_import_winbiz/wizard/l10n_ch_import_winbiz.py

```diff
@@ -0,0 +1,289 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+from xlrd import open_workbook, xldate_as_tuple
+import tempfile
+from openerp import models, fields, api, exceptions
+from openerp.tools.translate import _
+import datetime
+
+
+class AccountWinbizImport(models.TransientModel):
+    _name = 'account.winbiz.import'
+    _description = 'Import Accounting Winbiz'
+    _rec_name = 'state'
+
+    company_id = fields.Many2one('res.company', 'Company', invisible=True)
+    enable_account_based_line_merging = fields.Boolean(
+        "Group Balance Sheet accounts lines", default=False)
+    report = fields.Text('Report', readonly=True)
+    state = fields.Selection(selection=[
+        ('draft', "Draft"),
+        ('done', "Done"),
+        ('error', "Error")],
+        readonly=True,
+        default='draft')
+    file = fields.Binary('File', required=True)
+    imported_move_ids = fields.Many2many(
+        'account.move', 'import_winbiz_move_rel',
+        string='Imported moves')
+
+    def _parse_xls(self):
+        """Parse stored Excel file.
+
+        Manage base 64 decoding.
+
+        :param imp_id: current importer id
+        :returns: generator
+
+        """
+        # We use tempfile in order to avoid memory error with large files
+        with tempfile.TemporaryFile() as src:
+            content = self.file
+            src.write(content)
+            with tempfile.NamedTemporaryFile() as decoded:
+                src.seek(0)
+                base64.decode(src, decoded)
+                decoded.seek(0)
+                wb = open_workbook(decoded.name, encoding_override='cp1252')
+                self.wb = wb
+                sheet = wb.sheet_by_index(0)
+
+                def vals(n):
+                    return [c.value for c in sheet.row(n)]
+                head = vals(0)
+                res = [dict(zip(head, vals(i)))
+                       for i in xrange(1, sheet.nrows)]
+                res.sort(key=lambda e: e[u'numéro'])
+                return res
+
+    def _parse_date(self, date):
+        """Parse a date coming from Excel.
+
+           :param date: cell value
+           :returns: datetime.date
+        """
+        if isinstance(date, basestring):
+            from babel import Locale
+            d, m, y = date.split('-')
+            d = int(d)
+            mapping = Locale('en').months['format']['abbreviated']
+            mapping = dict(zip(mapping.values(), mapping.keys()))
+            m = mapping[m]
+            y = 2000 + int(y)
+            return datetime.datetime(y, m, d)
+        else:
+            return datetime.datetime(*xldate_as_tuple(date, self.wb.datemode))
+
+    @api.multi
+    def _standardise_data(self, data):
+        """
+        This function split one line of the spreadsheet into multiple lines.
+        Winbiz just writes one line per move.
+        """
+
```

Comment:
That's a huge function. Can't it be splitted ? Can't the inner class be declared outside ? what is the point of declaring it there ?